internal/provider/data_source_acccounts_test.go

```diff
@@ -0,0 +1,71 @@
+package provider
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/hashicorp/boundary/testing/controller"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var (
+	fooAccountDataMissingAuthMethodId = `
+data "boundary_accounts" "foo" {}
+`
+
+	fooAccountData = `
+data "boundary_accounts" "foo" {
+	auth_method_id = boundary_account.foo.auth_method_id
+}
+`
+)
+
+func TestAccDataSourceAccounts(t *testing.T) {
+	tc := controller.NewTestController(t, tcConfig...)
+	defer tc.Shutdown()
+	url := tc.ApiAddrs()[0]
+
+	var provider *schema.Provider
+
+	resource.Test(t, resource.TestCase{
+		ProviderFactories: providerFactories(&provider),
+		Steps: []resource.TestStep{
+			{
+				Config:      testConfig(url, fooAccountDataMissingAuthMethodId),
+				ExpectError: regexp.MustCompile("auth_method_id: Invalid formatted identifier."),
+			},
+			{
+				Config: testConfig(url, fooOrg, fooAccount, fooAccountData),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttrSet("data.boundary_accounts.foo", "auth_method_id"),
+					resource.TestCheckResourceAttr("data.boundary_accounts.foo", "items.#", "1"),
+					resource.TestCheckResourceAttr("data.boundary_accounts.foo", "items.0.%", "11"),
+					resource.TestCheckResourceAttrSet("data.boundary_accounts.foo", "items.0.auth_method_id"),
+					resource.TestCheckResourceAttr("data.boundary_accounts.foo", "items.0.authorized_actions.#", "6"),
```

Comment:
Love the detailed checking on all resource attributes here, including authorized actions.